luatest/server.lua

```diff
@@ -106,21 +181,53 @@ function Server:initialize()
     end
 end
 
---- Generates environment to run process with.
--- The result is merged into os.environ().
--- @return map
+--- Create a table with env variables based on the constructor params.
+-- The result will be passed into the server process.
+-- Table consists of the following entries:
+--
+--   * TARANTOOL_WORKDIR
+--   * TARANTOOL_LISTEN
+--   * TARANTOOL_ALIAS
+--   * TARANTOOL_HTTP_PORT
+--   * TARANTOOL_BOX_CFG
+--
+-- @return table
 function Server:build_env()
-    return {
+    local res = {
         TARANTOOL_WORKDIR = self.workdir,
-        TARANTOOL_HTTP_PORT = self.http_port,
         TARANTOOL_LISTEN = self.net_box_port or self.net_box_uri,
         TARANTOOL_ALIAS = self.alias,
     }
+    if self.http_port ~= nil then
+        res.TARANTOOL_HTTP_PORT = self.http_port
+    end
+    if self.box_cfg ~= nil then
+        res.TARANTOOL_BOX_CFG = json.encode(self.box_cfg)
+    end
+    return res
 end
 
---- Start server process.
-function Server:start()
+--- Build a listen URI based on the given server alias.
+-- For now, only UNIX sockets are supported.
+--
+-- @string server_alias Server alias.
+-- @return string
+function Server.build_listen_uri(server_alias)
+    return fio.pathjoin(Server.vardir, server_alias .. '.sock')
+end
+
+--- Start a server.
+-- Optionally waits until the server is ready.
```

Comment:
If the default behavior is to wait, it is more correct to say that it optionally doesn't wait.

Reply:
The default behavior is to wait unless a custom executable was provided while building the server object.

Reply:
What if I explicitly pass `true` and a custom executable?

Reply:
> What if I explicitly pass true and a custom executable?

The waiting will be on.

> Will the fact that there's a custom executable override the true value and make it not wait?

Yes.